contents/ruby/Rails.md

```diff
@@ -0,0 +1,154 @@
+
+In order to achieve this in the "Rails" way, it has a convenient abstraction called a "Resource". A resource refers to a RESTful resource that can be acted upon with the HTTP verbs (GET, POST, PUT, and DELETE), and can be Created, have its details Read, Updated, and Destroyed (CRUD). In order to declare a resource, you would mention it inside your `routes.rb` in the following manner:- 
+
+```ruby
+resources :users
+```
+
+Now, you can check all the routes that you have in your application by running `bin/rails routes` in your Rails CLI. This should include:
+
+```
+                   Prefix Verb   URI Pattern                                                                              Controller#Action
+                    users GET    /users(.:format)                                                                         users#index
+                          POST   /users(.:format)                                                                         users#create
+                 new_user GET    /users/new(.:format)                                                                     users#new
+                edit_user GET    /users/:id/edit(.:format)                                                                users#edit
+                     user GET    /users/:id(.:format)                                                                     users#show
+                          PATCH  /users/:id(.:format)                                                                     users#update
+                          PUT    /users/:id(.:format)                                                                     users#update
+                          DELETE /users/:id(.:format)                                                                     users#destroy
+```
+
+As can be seen, the common routes that you would need for the resources have already been mapped to URI's. Rails will then look for a controller called `users_controller.rb`, as is convention, and invoke the name of the function mapped to the particular URI and Verb. More information about Rails routes and how they work can be found [here](https://guides.rubyonrails.org/routing.html#resource-routing-the-rails-default).
+
+#### Controller
+
+In the standard Rails convention, once inside the application directory, if you navigate to `app/controllers/`, you should be able to see all the controllers in your project. Rails will look for the controller with the `<name>_controller.rb`, where `<name>` would be replaced by the result under the `Controller` heading in the list of routes above, in this case, "users". It will then look for the function name under the `Action` heading mapped to the URI function inside the Controller, and then call it. This is where the bulk of the business logic of your application would be. So if the person makes a GET request to `/users`, then Rails will open `users_controller.rb`, and call its `index` function.
```

Comment:
I think I get the rough idea, but I think it could be clearer if a sample controller file is available to guide the reader along! It should make the explanation simpler as well.

Reply:
Oh yeah that's true, ideally you can include sample files for the model, controller, and the index view - in a way that corresponds with the diagram you inserted.

Reply:
Interesting idea, I'll do that :) 

Screenshots perhaps?